### PR TITLE
Enhance ToggleGroup theme to allow divider to be removed and rounding on individual buttons

### LINKED ISSUE #7388: ToggleGroup — Add option to have rounding on all corners of individual buttons
Design: https://www.figma.com/design/drD2ft9yyKcj1UUOL0LSnC/Theme-update?node-id=8-5699&node-type=frame&t=it0GZeSaQgeF4ms1-11

Current implementation: https://deploy-preview-4256--keen-mayer-a86c8b.netlify.app/components/togglegroup

Deliverable:
- [ ] PR to Grommet with enhancement
- [ ] Update to theme in https://github.com/grommet/hpe-design-system/pull/4256

### LINKED ISSUE #7389: ToggleGroup — Add ability to set divider to "false"
Design: https://www.figma.com/design/drD2ft9yyKcj1UUOL0LSnC/Theme-update?node-id=8-5699&node-type=frame&t=it0GZeSaQgeF4ms1-11

Current implementation: https://deploy-preview-4256--keen-mayer-a86c8b.netlify.app/components/togglegroup

Deliverable:
- [ ] PR to Grommet with enhancement
- [ ] Update to theme in https://github.com/grommet/hpe-design-system/pull/4256

### PR DESCRIPTION
**Pull request by taysea**
#### What does this PR do?
This PR enhances ToggleGroup theme to allow:
1. The divider between individual toggle buttons to be removed by setting `toggleGroup.divider` to false.
2. Rounding on individual buttons to be defined via `toggleGroup.button.border.radius`
3. Padding only to be adjusted if `theme.button.intelligentPad = true`

These enhancements are in alignment with prep for design tokens.
 
#### Where should the reviewer start?
src/js/components/ToggleGroup/ToggleGroup.js

#### What testing has been done on this PR?

Locally by adjusting `preview.js`:

```
import React, { useState, useEffect } from 'react';
import Root from 'react-shadow';
import { StyleSheetManager } from 'styled-components';
import { hpe } from 'grommet-theme-hpe';
import isChromatic from 'chromatic/isChromatic';
import { Grommet, grommet, hacktoberfest2022, Box, Text } from '../src/js';
import { deepMerge } from '../src/js/utils';

const themeUpdate = deepMerge(hpe, {
  toggleGroup: {
    divider: false,
    button: {
      border: {
        radius: 'xsmall',
      },
    },
  },
});

const CUSTOM_THEMED = 'Custom Themed';
const THEMES = {
  hpe: themeUpdate,
  grommet,
  hacktoberfest2022,
  base: {},
};

export const decorators = [
  (Story, context) => {
    const [rootRef, setRootRef] = useState(null);
    const [state, setState] = useState('grommet');
    const [root, setRoot] = useState('document');
    const full = context.allArgs?.full || 'min';
    const dir = context.allArgs?.dir;
    const options = context.allArgs?.options;

    useEffect(() => {
      setState(context.globals.theme);
    }, [context.globals.theme]);
    useEffect(() => {
      setRoot(context.globals.root);
    }, [context.globals.root]);

    /**
     * This demonstrates that custom themed stories are driven off the "base"
     * theme. Custom themed stories will live under a "CustomThemed" directory.
     */
    if (context.kind.split('/')[2] === CUSTOM_THEMED && state !== 'base') {
      // if we are running the story in chromatic we want the chromatic snapshot
      // to be taken in the base theme for custom theme stories
      if (isChromatic()) {
        return (
          <Grommet theme={THEMES.base}>
            <Story state={THEMES.base} />
          </Grommet>
        );
      }
      return (
        <Box align="center" pad="large">
          <Text size="large">
            {`Custom themed stories are only displayed in the
                "base" theme mode. To enable, select "base" from the
                Theme menu above.`}
          </Text>
          <div hidden>
            <Story state={THEMES[state]} />
          </div>
        </Box>
      );
    }

    if (root === 'shadow') {
      return (
        // eslint-disable-next-line react/jsx-pascal-case
        <Root.div ref={setRootRef}>
          {rootRef && (
            <StyleSheetManager target={rootRef.shadowRoot}>
              <Grommet
                theme={THEMES[state]}
                full={full}
                dir={dir}
                options={options}
                containerTarget={rootRef.shadowRoot}
              >
                <Story state={THEMES[state]} />
              </Grommet>
            </StyleSheetManager>
          )}
        </Root.div>
      );
    }

    return (
      <Grommet theme={THEMES[state]} full={full} dir={dir} options={options}>
        <Story state={THEMES[state]} />
      </Grommet>
    );
  },
];

export const parameters = {
  layout: 'fullscreen',
  options: {
    storySort: (first, second) => {
      /**
       * The story sort algorithm will only ever compare two stories
       * a single time. This means that every story will only ever be either
       * the "first" parameter OR the "second" parameter, but not both.
       * So, the checks for custom themed stories need to happen on both inputs
       * of this function.
       *
       * A return value of 1 results in sorting the "first" story AFTER the
       * "second" story.
       *
       * A return value of 0 results in sorting the "first" story BEFORE the
       * secondary story.
       */
      const isFirstCustom = first[1].kind.split('/')[2] === CUSTOM_THEMED;
      const isSecondCustom = second[1].kind.split('/')[2] === CUSTOM_THEMED;
      if (isFirstCustom) return 1;
      if (isSecondCustom) return 0;
      return first[1].kind === second[1].kind
        ? 0
        : first[1].id.localeCompare(second[1].id, undefined, { numeric: true });
    },
  },
};

export const globalTypes = {
  theme: {
    defaultValue: 'grommet',
    toolbar: {
      title: 'Theme',
      items: [
        { title: 'base', value: 'base' },
        { title: 'grommet', value: 'grommet' },
        { title: 'hpe', value: 'hpe' },
        { title: 'hacktoberfest2022', value: 'hacktoberfest2022' },
      ],
    },
  },
  root: {
    defaultValue: 'document',
    toolbar: {
      title: 'Root',
      items: ['document', 'shadow'],
    },
  },
};

```

#### Do Jest tests follow these best practices?

- [ ] `screen` is used for querying.
- [ ] The correct query is used. (Refer to [this list of queries](https://testing-library.com/docs/queries/about/#priority))
- [x] `asFragment()` is used for snapshot testing.

#### What are the relevant issues?
Closes #7389 
Closes #7388 

#### Screenshots (if appropriate)

Screenshot 2024-10-21 at 1 52 22 PM

#### Do the grommet docs need to be updated?
Yes.

#### Should this PR be mentioned in the release notes?
Yes.

#### Is this change backwards compatible or is it a breaking change?
Backwards compatible.

### CODE REVIEW

**Review by MikeKingdom — APPROVED** [review]
LGTM

File: src/js/themes/base.js [hunk 1]
```diff
@@ -1948,6 +1948,9 @@ export const generate = (baseSpacing = 24, scale = 6) => {
           horizontal: '11px',
           vertical: '11px',
         },
+        // border: {
```
Comment (halocline) [inline src/js/themes/base.js:1951]: Alphabetize order of properties for code style?

File: src/js/themes/base.d.ts [hunk 2]
```diff
@@ -1734,11 +1734,16 @@ export interface ThemeType {
       iconOnly?: {
         pad?: PadType;
       };
+      border?: {
```
Comment (halocline) [inline src/js/themes/base.d.ts:1737]: Alphabetize order of properties for code style?

**Review by halocline — COMMENTED** [review]
Two small code style conventions nits, otherwise looks good.

File: src/js/themes/base.d.ts (hunk 2 above)
Comment (taysea) [reply src/js/themes/base.d.ts:1733]: Done.

File: src/js/themes/base.js (hunk 1 above)
Comment (taysea) [reply src/js/themes/base.js]: Done.

**Review by halocline — APPROVED** [review]

**Pull request merged**